cmd/bnsd/app/datamigration.go

```diff
@@ -44,4 +51,97 @@ func init() {
 			return nil
 		},
 	})
+
+	datamigration.MustRegister("rewrite username accounts", datamigration.Migration{
+		RequiredSigners: []weave.Address{governingBoard},
+		ChainIDs: []string{
+			"iov-mainnet",
+		},
+		Migrate: rewriteUsernameAccounts,
+	})
 }
+
+func rewriteUsernameAccounts(ctx context.Context, db weave.KVStore) error {
+	governingBoard, err := weave.ParseAddress("cond:gov/rule/0000000000000001")
+	if err != nil {
+		panic(err)
+	}
+
+	now, err := weave.BlockTime(ctx)
+	if err != nil {
+		return errors.Wrap(err, "block time")
+	}
+
+	iov := account.Domain{
+		Metadata:     &weave.Metadata{Schema: 1},
+		Domain:       "iov",
+		Admin:        governingBoard,
+		HasSuperuser: false,
+		AccountRenew: weave.AsUnixDuration(30 * 24 * time.Hour),
+		// IOV domain is not supposed to expire. It is
+		// not our problem in 100 years ;)
+		ValidUntil: weave.AsUnixTime(now.Add(oneHundredYears)),
+	}
+	if _, err := account.NewDomainBucket().Put(db, []byte("iov"), &iov); err != nil {
+		return errors.Wrap(err, "save iov domain")
+	}
+
+	accounts := account.NewAccountBucket()
+
+	// Every domain must contain an empty account.
+	empty := &account.Account{
+		Metadata:     &weave.Metadata{Schema: 1},
+		Domain:       "iov",
+		Name:         "",
+		Owner:        iov.Admin,
+		ValidUntil:   weave.AsUnixTime(now.Add(iov.AccountRenew.Duration())),
```

Comment:
Because we do not have an information when the token was created, each account is getting refreshed on migration. I think this is not an issue.